site/docs/snapshot-tag-branch.md

```diff
@@ -0,0 +1,168 @@
+<!--
+ - Licensed to the Apache Software Foundation (ASF) under one or more
+ - contributor license agreements.  See the NOTICE file distributed with
+ - this work for additional information regarding copyright ownership.
+ - The ASF licenses this file to You under the Apache License, Version 2.0
+ - (the "License"); you may not use this file except in compliance with
+ - the License.  You may obtain a copy of the License at
+ -
+ -   http://www.apache.org/licenses/LICENSE-2.0
+ -
+ - Unless required by applicable law or agreed to in writing, software
+ - distributed under the License is distributed on an "AS IS" BASIS,
+ - WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ - See the License for the specific language governing permissions and
+ - limitations under the License.
+ -->
+
+# Snapshot Tagging and Branching
+
+Iceberg snapshot tagging and branching feature provides users more functionalities in managing Iceberg snapshot lifecycle.
+Users can assign tags to snapshots, create new branches, set the current branch for read and write, and configure customized retention policy for them.
+
+## Example use cases
+
+### Time-based Snapshot tagging
```

Comment:
It isn't clear from the spec how this would be done, especially without the `max-ref-age-ms` property.

Reply:
My original thinking is that if the snapshot is expired and it has reference, the reference is removed at the same time. But I agree `max-ref-age-ms` does give more flexibility, I will update based on that.